oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/OAuth2AuthorizationServerMetadataEndpointFilter.java

```diff
@@ -83,6 +83,7 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 				.tokenEndpoint(asUrl(this.providerSettings.getIssuer(), this.providerSettings.getTokenEndpoint()))
 				.tokenEndpointAuthenticationMethods(clientAuthenticationMethods())
 				.jwkSetUrl(asUrl(this.providerSettings.getIssuer(), this.providerSettings.getJwkSetEndpoint()))
+				.userInfoEndpoint(asUrl(this.providerSettings.getIssuer(), this.providerSettings.getOidcUserInfoEndpoint()))
```

Comment:
I think the preference at the moment would be to remove this from `OAuth2AuthorizationServerMetadataEndpointFilter` at this time, as we want to align as closely to the spec as possible (and not support optional features/items such as this).

Since the `OAuth2AuthorizationServerMetadataEndpointFilter` is not very customizable at the moment, we should see if anyone requires this, and open an enhancement for customization options instead if needed.

Would you be able to provide an update to this PR to address this?